File: QubitServer/src/org/labrad/qubits/channeldata/AnalogDataBase.java
package org.labrad.qubits.channeldata;

import org.labrad.qubits.channels.AnalogChannel;

public abstract class AnalogDataBase implements AnalogData {

  private AnalogChannel channel;

  private boolean isDeconvolved = false;

  @Override
  public void setChannel(AnalogChannel channel) {
    this.channel = channel;
  }

  protected AnalogChannel getChannel() {
    return channel;
  }

  /**
   * Whether this bit of analog data has been deconvolved.
   */
  @Override
  public boolean isDeconvolved() {
    return isDeconvolved;
  }

  @Override
  public void invalidate() {
    isDeconvolved = false;
  }
  
  protected void setDeconvolved(boolean isDeconvolved) {
    this.isDeconvolved = isDeconvolved;
  }
}

File: QubitServer/src/main/java/org/labrad/qubits/channeldata/AnalogDataBase.scala
package org.labrad.qubits.channeldata

import org.labrad.qubits.channels.AnalogChannel

abstract class AnalogDataBase extends AnalogData {

  private var channel: AnalogChannel = null

  @volatile private var _isDeconvolved = false

  override def setChannel(channel: AnalogChannel): Unit = {
    this.channel = channel
  }

  protected def getChannel(): AnalogChannel = {
    channel
  }

  /**
   * Whether this bit of analog data has been deconvolved.
   */
  override def isDeconvolved(): Boolean = {
    _isDeconvolved
  }

  override def invalidate(): Unit = {
    setDeconvolved(false)
  }

  private[channeldata] def setDeconvolved(isDeconvolved: Boolean): Unit = {
    this._isDeconvolved = isDeconvolved
  }
}
